scripts/stablex/utilities.js

```diff
@@ -35,34 +48,45 @@ const fetchTokenInfo = async function(
   return tokenObjects
 }
 
-const addTokens = async function(token_addresses, web3, artifacts) {
-  const accounts = await web3.eth.getAccounts()
-
-  const BatchExchange = artifacts.require("BatchExchange")
-  const instance = await BatchExchange.deployed()
-
-  const TokenOWL = artifacts.require("../node_modules/@gnosis.pm/owl-token/build/contracts/TokenOWL")
-  const owl = await TokenOWL.at(await instance.feeToken.call())
+const addTokens = async function({ tokenAddresses, account, batchExchange, owl }) {
+  // Get amount of required OWL for listing all tokens
+  const feeForAddingToken = await batchExchange.FEE_FOR_LISTING_TOKEN_IN_OWL.call()
+  const totalFees = feeForAddingToken.mul(new BN(tokenAddresses.length))
 
-  const feeForAddingToken = (await instance.FEE_FOR_LISTING_TOKEN_IN_OWL.call()).mul(new BN(token_addresses.length))
-  const balanceOfOWL = await owl.balanceOf.call(accounts[0])
-  if (feeForAddingToken.gt(balanceOfOWL)) {
-    console.log("More fee tokens are required to add all tokens")
-    return
+  // Ensure the user has enough OWL balance
+  const balanceOfOWL = await owl.balanceOf.call(account)
+  if (totalFees.gt(balanceOfOWL)) {
+    throw new Error("More fee tokens are required to add all tokens")
```

Comment:
@fleupold @josojo I just want to double check that this is fine with you guys.
I hope I don't break anything with this. I just thought this is an error that should be raised instead of console logged.

Reply:
Yes, I agree that's better.